modules/gridmap/grid_map_editor_plugin.cpp

```diff
@@ -455,6 +462,29 @@ void GridMapEditor::_delete_selection() {
 	_validate_selection();
 }
 
+void GridMapEditor::_fill_selection() {
+
+	if (!selection.active)
+		return;
+
+	undo_redo->create_action(TTR("GridMap Fill Selection"));
+	for (int i = selection.begin.x; i <= selection.end.x; i++) {
+
+		for (int j = selection.begin.y; j <= selection.end.y; j++) {
+
+			for (int k = selection.begin.z; k <= selection.end.z; k++) {
+
+				undo_redo->add_do_method(node, "set_cell_item", i, j, k, selected_pallete, cursor_rot);
```

Comment:
Wouldn't one expect this kind of bucket fill to only fill the empty cells, and not override the existing ones? (as the bucket fill does for 2D tilemaps)

Reply:
A bucket fill tool would also be useful, but that's not what this tool is for,  This tool exists for quickly blocking walls and floors.  It is also useful for erasing undesired details from a floor or wall, returning the floor or wall to a plain state.  For the latter functionality, it is absolutely necessary that it overwrites all cells.